controller/telemetry/server.go

```diff
@@ -399,13 +398,24 @@ func convertSampleStream(sample *model.SampleStream) *read.Sample {
 }
 
 func (s *server) requestLabelsFor(requestScope *write.RequestScope) prometheus.Labels {
-	sourceName, sourceDeployment := s.getNameAndDeployment(requestScope.Ctx.SourceIp)
-	targetName, targetDeployment := s.getNameAndDeployment(requestScope.Ctx.TargetAddr.Ip)
+	var sourceDeployment string
+	sourceDeploymentPtr, _ := s.getDeployment(requestScope.Ctx.SourceIp)
+	if sourceDeploymentPtr == nil {
```

Comment:
If we always replace nils with empty strings, why not have `getDeployments` just return empty strings in the first place?